frontend/src/components/button.rs

```diff
@@ -2,17 +2,18 @@ use yew::prelude::*;
 
 #[derive(Properties, PartialEq)]
 pub struct ButtonProps {
-    pub name: String,
+    pub children: Children,
```

Comment:
no need to carry around the children in props only for the icon. We can use an Option<iconID> for this